### PR TITLE
Encoding

### PR DESCRIPTION
**Pull request by a-frantz**
This has been sitting for a few months because I couldn't find fastqs in the alternate encodings for test purposes. I think if we put in the docs that only `sanger` encoding has been thoroughly tested that might be ok. No idea where to get my hands on the others unless we're ok with me mocking them from scratch for test purposes.

Figure there's no sense in letting the branch continue to sit as I've stopped looking for real data in the alternate encodings.

Detecting is very straight-forward, either the PHRED scores are in the set of possible values or they aren't. For the purposes of this module, don't see any pitfalls of using mocked data.

### CODE REVIEW

**Review by adthrasher — APPROVED** [review]
It may be worth pointing out in the help documentation that if there are no low quality reads, this won't be able to detect (which may not matter). Also if there is no low quality read in the n reads that are sampled, it won't be able to detect with certainty.

File: ngsderive/commands/encoding.py
Comment (adthrasher) [inline ngsderive/commands/encoding.py:12]: Should probably lower the default logging verbosity before merging.

**Comment by a-frantz** [comment]
Yes that would be good to point out. Also I think the only reason we even care is because it's a parameter in `fastq_screen`. I haven't figured out what FastQ-Screen uses it for, but as long as it's within a set that won't cause it to error, I don't think we care.

**Review by claymcleod — APPROVED** [review]

File: ngsderive/utils.py
Comment (claymcleod) [inline ngsderive/utils.py:72]: I might conditionally decode this is store qualities is set to true. Else you do that work even if the quality will never  be used.

Comment (claymcleod) [reply ngsderive/utils.py:72]: This will very likely get you much more performance benefit that what the conditional below would do (simply exclude it from the yielded dictionary).

**Comment by claymcleod** [comment]
This needs an addition to the docs page with, at minimum, links to the information on the various phred score ranges. Otherwise looks good. I did not test it, just reviewed the code.

**Comment by a-frantz** [comment]
@claymcleod @adthrasher I'm going to merge this unless either of you think BAM support is needed. Not sure if it can be easily done while using `pysam`, because pysam automatically decodes the PHRED scores (from the docs it seems it assumes the Sanger/Illumina1.8+ encoding. It's the most common and also what all our files use).

**Pull request merged**
